src/impls/scale.rs

```diff
@@ -0,0 +1,304 @@
+use crate::U256;
+use derive_more::{From, Into};
```

Comment:
I don't think its worth pulling in and building a full proc-macro dependency for a couple `From` and `Into` implementations.

Reply:
What would be the benefit of not using proc macros here?

Reply:
I think in both cases, the benefits and downsides are very marginal. The reason I'm not a fan of it is:
- Adds an additional dependency for 10 lines of boilerplate and error-free code (as in, you can't really make a mistake in those lines that the compiler won't catch).
- AFAICT, it is `impl Into<U256> for CompactU256` instead of the preferred `impl From<CompactU256> for U256`. If this is not the case (i.e. the proc-macro is implementing the reverse `From` implementation when you derive `Into`) then I would argue that the code isn't as clear as it could be.
- Slightly slower build times. On my M1 Pro I clocked it at 2.5 additional seconds per fresh build. While it may not be significant for larger projects, for smaller ones this is a non-trivial amount of time.
- And a slightly more concrete reason: I've had issues in the past in building statically linked binaries when using proc macros. I know that it has improved recently, but I _think_ there still may be an issue if you are cross compiling to a musl target for example (not 100% sure).

All in all, I recognize that the arguments against are quite weak. However, I think the benefit of using this extra dependency is so marginal that I think it is better to manually implement `From` even if the arguments for that are weak.

Reply:
**edit**: For issues with proc macros on musl I was able to reproduce with:
```
% cd some/project/with/derive_more/dependency
% docker run -it --rm -v "$(pwd):/src" -w /src rust:alpine cargo build
...
   Compiling derive_more v0.99.17
error: linking with `cc` failed: exit status: 1
...
```

Reply:
Thank you for detailed explanation. Replaced with manual impls.